lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -4124,3 +4135,28 @@ instance FromJSON (ApiT TxScriptValidity) where
 instance ToJSON (ApiT TxScriptValidity) where
     toJSON = genericToJSON Aeson.defaultOptions
         { constructorTagModifier = camelTo2 '_' . drop 8 } . getApiT
+
+--------------------------------------------------------------------------------
+-- Utility types
+--------------------------------------------------------------------------------
+
+-- | A wrapper that allows any type to be serialized as a JSON array.
+--
+-- The number of items permitted in the array is dependent on the wrapped type.
+--
+newtype ApiAsArray a = ApiAsArray a
+    deriving (Eq, Generic, Show, Typeable)
+    deriving newtype (Monoid, Semigroup)
+    deriving anyclass NFData
+
+instance FromJSON a => FromJSON (ApiAsArray (Maybe a)) where
+    parseJSON json = parseJSON @[a] json >>= \case
+        [a] ->
+            pure $ ApiAsArray $ Just a
+        [] ->
+            pure $ ApiAsArray Nothing
+        _  ->
+            fail "Expected at most one item."
```

Comment:
Suggestion: Improve this error message to make it more useful for the user.

It could be more useful if we were able to report something like:

> Expected at most one item within "collateral_outputs".

Reply:
Fixed in 3cd2c4e99cf9ba9b228ae0b2d344d63cbd7a2aa6.